### ISSUE TITLE
Cosmos DB "-server" suffix

### ISSUE THREAD

**Issue by JordanMarr** [CompositionalIT/farmer#156]
Having a great time with Farmer today creating my first template!

One small issue is that my Cosmos DB instance has "-server" appended to the name.  
I specified this:
`name "jmarr-appname-db"` 

But it shows up in the Azure Portal as "jmarr-appname-db-server".

Here is an excerpt from the generated template:

```
    {
      "apiVersion": "2020-03-01",
      "kind": "GlobalDocumentDB",
      "location": "eastus",
      "name": "jmarr-appname-db-server",
      "properties": {
        "consistencyPolicy": {
          "defaultConsistencyLevel": "Eventual"
        },
        "databaseAccountOfferType": "Standard",
        "enableFreeTier": true,
        "publicNetworkAccess": "Enabled"
      },
      "tags": {
        "CosmosAccountType": "Non-Production",
        "defaultExperience": "Core (SQL)"
      },
      "type": "Microsoft.DocumentDB/databaseAccounts"
    },
    {
      "apiVersion": "2020-03-01",
      "dependsOn": [
        "jmarr-appname-db-server"
      ],
      "name": "jmarr-appname-db-server/jmarr-appname-db",
      "properties": {
        "options": {
          "throughput": "400"
        },
        "resource": {
          "id": "jmarr-appname-db"
        }
      },
      "type": "Microsoft.DocumentDB/databaseAccounts/sqlDatabases"
    },
```

**Comment by isaacabraham** [comment]
Use the `account_name` keyword in the builder :-) `name` specifies the name of the database. This is perhaps a little unclear - how could we make it more obvious?

**Comment by JordanMarr** [comment]
Ah ok, I see my mistake.

The "problem" is that Farmer let me move so quickly that I started to get disconnected from the Azure Cosmos DB domain nouns.  The ability to move so quickly is actually a really nice problem to have.  :)
After taking a step back and reviewing the Azure Portal "Create" Cosmos DB page, I am reminded that a Cosmos "account" can have multiple "databases"; my intent was to simplify this by sharing the same between the account and the database.  

So in other words, this is reality:

- Account
  - Database1
    - ContainerA
    - ContainerB
  - Database2
    - ContainerC
    - ContainerD

From Farmer's perspective, the you start at the _database_ level via the `cosmosDb` builder and specify the `account_name` rather than starting at the _account_ level via a `cosmosAccount` builder.  
I suppose the alternative would be this:
```
let myCosmosAccount = cosmosAccount { 
    name "jmarr-appname-db-server"
    add_db [ 
        cosmosDb { 
            name "jmarr-appname-db"
            // ...
            add_containers [
                // ...
            ]
        }
    ]
}
```

I think your current design is probably better because it results in less code.  

I see two simple documentation fixes:
1. Fix "server_name" -> should be "account_name"
2. Fix "Server" text -> should be "Account"
3. Fix "link_to_server" -> should be "link_to_account"

Incoming documentation PR...

**Comment by isaacabraham** [comment]
@JordanMarr yeah, I fixed `server_name` in the latest release but probably forget the others!! And yes, a design decision I made was to "collapse" the notion of DB and server into one to support people getting starter that they only need to know one thing. I'm still not convinced it's the right solution. It applies not just to Cosmos but in several other places: SQL, Service Bus Queues, Web Apps - essentially anywhere that there's a logical "container" with a many-children relationship. We could probably make another builder that does exactly what you're thinking of - as an alternative to the "folded" model.
